specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json

```diff
@@ -551,31 +550,395 @@
+                },
+                {
+                  "name": "ReadHsmBackupStatus",
+                  "value": "Microsoft.KeyVault/managedHsm/backup/status/action",
+                  "description": "Read HSM backup status."
+                },
+                {
+                  "name": "ReadHsmRestoreStatus",
+                  "value": "Microsoft.KeyVault/managedHsm/restore/status/action",
+                  "description": "Read HSM restore status."
+                }
+              ]
+            }
+          }
         },
         "notDataActions": {
           "type": "array",
           "items": {
-            "type": "string"
-          },
-          "description": "Denied Data actions."
+            "type": "string",
+            "description": "Denied Data actions.",
+            "enum": [
+              "Microsoft.KeyVault/managedHsm/keys/read/action",
+              "Microsoft.KeyVault/managedHsm/keys/write/action",
+              "Microsoft.KeyVault/managedHsm/keys/deletedKeys/read/action",
+              "Microsoft.KeyVault/managedHsm/keys/deletedKeys/recover/action",
+              "Microsoft.KeyVault/managedHsm/keys/backup/action",
+              "Microsoft.KeyVault/managedHsm/keys/restore/action",
+              "Microsoft.KeyVault/managedHsm/roleAssignments/delete/action",
+              "Microsoft.KeyVault/managedHsm/roleAssignments/read/action",
+              "Microsoft.KeyVault/managedHsm/roleAssignments/write/action",
+              "Microsoft.KeyVault/managedHsm/roleDefinitions/read/action",
+              "Microsoft.KeyVault/managedHsm/keys/encrypt/action",
+              "Microsoft.KeyVault/managedHsm/keys/decrypt/action",
+              "Microsoft.KeyVault/managedHsm/keys/wrap/action",
+              "Microsoft.KeyVault/managedHsm/keys/unwrap/action",
+              "Microsoft.KeyVault/managedHsm/keys/sign/action",
+              "Microsoft.KeyVault/managedHsm/keys/verify/action",
+              "Microsoft.KeyVault/managedHsm/keys/create",
+              "Microsoft.KeyVault/managedHsm/keys/delete",
+              "Microsoft.KeyVault/managedHsm/keys/export/action",
+              "Microsoft.KeyVault/managedHsm/keys/import/action",
+              "Microsoft.KeyVault/managedHsm/keys/deletedKeys/delete",
+              "Microsoft.KeyVault/managedHsm/securitydomain/download/action",
+              "Microsoft.KeyVault/managedHsm/securitydomain/upload/action",
+              "Microsoft.KeyVault/managedHsm/securitydomain/upload/read",
+              "Microsoft.KeyVault/managedHsm/securitydomain/transferkey/read",
+              "Microsoft.KeyVault/managedHsm/backup/start/action",
+              "Microsoft.KeyVault/managedHsm/restore/start/action",
+              "Microsoft.KeyVault/managedHsm/backup/status/action",
+              "Microsoft.KeyVault/managedHsm/restore/status/action"
+            ],
+            "x-ms-enum": {
+              "name": "DeniedDataActions",
```

Comment:
Are these the same? If so, I would just call it DataActions and define a single enum (both for swagger and SDK source). Whether they are allowed or denied is separate from what they represent. At a glance, these seem to be the same (same number of items, and - spot checking - same items).

Reply:
they are the same, the enum can for sure be collapsed

also the docs should be changed here as NotDataActions are separate from deny assigments, which we don't support. NotDataActions and NotActions follow the same behavior as in central RBAC, where they are simply subtracted from the set of roles granted by a particular definition

Reply:
@Lusitanian Thanks for clarifying this - we used the deny naming in the SDK clients also based on this assumption. 

Given the "not" logic you describe, I'm not clear on how these would be used. Would it be for the purpose of assigning a "subtractive role" to someone that also had a more permissive role? For example, I want someone to be an admin minus the backup permission?

Reply:
it's basically for creating a role definition that includes a huge colleciton of permissions in "DataActions" and then subtracting one using a "NotDataAction" e.g. I grant "wrapKey, unwrapKey,  sign" but  take away "sign." by adding it as a NotDataAction. More info is here: https://docs.microsoft.com/en-us/azure/role-based-access-control/role-definitions 

I do not think it'll see much use but we wanted wire-compat with central RBAC's protocol